fcsparser/api.py

```diff
@@ -418,78 +460,125 @@ def get_channel_names(self):
 
     def version(self):
         """ Returns the fcs version from the TEXT """
-        return self.annotation['__header__']['FCS format']
+        return self.annotation["__header__"]["FCS format"]
 
     def _header_to_string(self):
         """
         Outputs the heads to a text format suitable for writing to an fcs file
         """
-        self.header_space_characters = b'    '
+        self.header_space_characters = b"    "
         self.header_offset_block_length = 8
 
         # set default (other methods rely on these having values)
         self.text_offset_start = 58
         self.text_offset_end = self.text_offset_start + 1
         self.data_offset_start = self.text_offset_end + 1
         self.data_offset_end = self.data_offset_start + 1
+
         self.analysis_offset_start = 0
         self.analysis_offset_end = 0
 
         # set actual
         self.text_offset_start = 58
-        self.text_offset_end = self.text_offset_start + len(self._annotation_to_string()) + 1
+        self.text_offset_end = (
+            self.text_offset_start + len(self._annotation_to_string()) + 1
+        )
         self.data_offset_start = self.text_offset_end + 1
-        self.data_offset_end = self.data_offset_start + len(self._data_to_byte_string()) - 2
+        self.data_offset_end = (
+            self.data_offset_start + len(self._data_to_byte_string()) - 2
+        )
         self.analysis_offset_start = 0
         self.analysis_offset_end = 0
 
-        return b''.join([self.version(),
-                         self.header_space_characters,
-                         str(self.text_offset_start).rjust(self.header_offset_block_length).encode('utf-8'),
-                         str(self.text_offset_end).rjust(self.header_offset_block_length).encode('utf-8'),
-                         str(self.data_offset_start).rjust(self.header_offset_block_length).encode('utf-8'),
-                         str(self.data_offset_end).rjust(self.header_offset_block_length).encode('utf-8'),
-                         str(self.analysis_offset_start).rjust(self.header_offset_block_length).encode('utf-8'),
-                         str(self.analysis_offset_end).rjust(self.header_offset_block_length).encode('utf-8'),
-                         BYTE_SEP])
+        # check for data start/end outside of maximum
```

Comment:
This is where the relevant changes begin

Reply:
Are there any other changes we should merge from upstream?

Reply:
There are a few in the original repo, but I didn't want to risk breaking anything.